src/Build/BackEnd/Components/Communications/NodeEndpointInProc.cs

```diff
@@ -332,11 +333,8 @@ private void EnqueuePacket(INodePacket packet)
             ErrorUtilities.VerifyThrow(null != _packetQueue, "packetQueue is null");
             ErrorUtilities.VerifyThrow(null != _packetAvailable, "packetAvailable is null");
 
-            lock (_packetQueue)
```

Comment:
I'm not sure if it matters since `Enqueue()` is blocking but `_packetAvailable.Set()` used to be in a `lock`.  Do you know if calling `Set()` in this manner requires some thread synchronization?

Reply:
According to documentation https://msdn.microsoft.com/en-us/library/system.threading.eventwaithandle.set(v=vs.110).aspx it doesn't. Also in my imagination of implementation such even it also doesn't.